manifests/otel-agent-reconciler-cm.yaml

```diff
@@ -0,0 +1,74 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This ConfigMap is used in the reconciler pod.
+# It contains the OpenTelemetry (OTEL) agent configuration for metrics and resource detection, with sync specific attributes.
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: otel-agent-reconciler
+  namespace: config-management-system
+  labels:
+    app: opentelemetry
+    component: otel-agent
+    configmanagement.gke.io/system: "true"
+    configmanagement.gke.io/arch: "csmr"
+data:
+  otel-agent-reconciler-config.yaml: |
+    receivers:
+      opencensus:
+        endpoint: 0.0.0.0:55678
+    exporters:
+      opencensus:
+        endpoint: otel-collector.config-management-monitoring:55678
+        tls:
+          insecure: true
+    processors:
+      # Attributes processor adds custom configsync metric labels to applicable
+      # metrics to identify the sync object used to configure this deployment.
+      #
+      # Note: configsync.sync.generation is explicitly excluded here, because it
+      # is high cardinality. So we don't want to send it as a label, only as a
+      # resource attribute. That way it's only propagated to Prometheus, and not
+      # Monarch or Cloud Monitoring, which ignore custom resource attributes.
+      attributes:
+        actions:
+          - key: configsync.sync.kind
+            action: upsert
+            value: ${CONFIGSYNC_SYNC_KIND}
```

Comment:
Did you verify these are getting replaced correctly?

IIRC, they were previously being injected using the downward API from reconciler Deployment metadata values, which are set by the reconciler-manager.

Reply:
I might be thinking of these ones: https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/manifests/templates/reconciler-manager-configmap.yaml#L242-L256

How do the variables here get replaced?

Reply:
Oh, I see. This ConfigMap is mounted on the otel-agent. So otel-collector must detect that they're environment variables and render them from the otel-agent container's environment variables, which are set on the reconciler Deployment I linked.

Reply:
Looks like we don't have test coverage for this, but in chat you said you verified manually. So it doesn't need to block. Please add tests for these metrics fields in another PR.